Winleafs.Wpf/Api/Events/EventTriggersCollection.cs

```diff
@@ -23,25 +37,78 @@ public EventTriggersCollection(Orchestrator orchestrator)
                     {
                         case TriggerType.ProcessEvent:
                             var processEventTrigger = (Models.Models.Scheduling.Triggers.ProcessEventTrigger)eventTrigger;
-                            EventTriggers.Add(new ProcessEventTrigger(eventTrigger, orchestrator, processEventTrigger.ProcessName, processEventTrigger.EffectName, processEventTrigger.Brightness));
+                            _eventTriggers.Add(new ProcessEventTrigger(this, processEventTrigger));
                             break;
 
-                        case TriggerType.Borderlands2HealthEvent:
+                        case TriggerType.SpotifyEvent:
+                            var spotifyEventTrigger = (Models.Models.Scheduling.Triggers.SpotifyEventTrigger)eventTrigger;
+                            _eventTriggers.Add(new SpotifyEventTrigger(this, spotifyEventTrigger));
+                            break;
+                        
+                        //Currently not in use
+                        /*case TriggerType.Borderlands2HealthEvent:
                             //This will never be reached currently, since users cannot add this type of event yet
                             EventTriggers.Add(new Borderlands2HealthEventTrigger(eventTrigger, orchestrator));
-                            break;
+                            break;*/
                     }
                 }
-            }            
-            
+            }
         }
 
-        public void StopAllEvents()
+        public async Task ActivateTrigger(int priority)
         {
-            foreach (var eventTrigger in EventTriggers)
+            if (!_activeTriggersPriorities.Contains(priority))
             {
-                eventTrigger.StopEvent();
+                _activeTriggersPriorities.Add(priority);
             }
+
+            await SetOperationMode();
+        }
+
+        public async Task DeactivateTrigger(int priority)
+        {
+            if (_activeTriggersPriorities.Remove(priority))
+            {
+                await SetOperationMode();
+            }
+        }
+
+        private async Task SetOperationMode()
+        {
+            if (_activeTriggersPriorities.Any())
+            {
+                var lowestPriority = _activeTriggersPriorities.Min();
+                if (ActiveTrigger == null || ActiveTrigger.Priority != lowestPriority)
```

Comment:
Reverse this if statement and return

Reply:
Not that straight forward because of the if else. I think it is currently more readable